src/gui/C4StartupScenSelDlg.cpp

```diff
@@ -1442,6 +1442,21 @@ C4StartupScenSelDlg::C4StartupScenSelDlg(bool fNetwork) : C4StartupDlg(LoadResSt
 	pScenSelCaption = new C4GUI::Label("", caBookLeft.GetFromTop(rScenSelCaptionFont.GetLineHeight()), ACenter, ClrScenarioItem, &rScenSelCaptionFont, false);
 	pSheetBook->AddElement(pScenSelCaption);
 	pScenSelCaption->SetToolTip(LoadResStr("IDS_DLGTIP_SELECTSCENARIO"));
+
+	// search bar
+	const char *labelText = LoadResStr("IDS_DLG_SEARCH");
+	int32_t width = 100, height;
+	::GraphicsResource.TextFont.GetTextExtent(labelText, width, height, true);
+	C4GUI::ComponentAligner caSearchBar(caBookLeft.GetFromBottom(height), 0, 0);
+	auto *searchLabel = new C4GUI::WoodenLabel(labelText, caSearchBar.GetFromLeft(width + 10), C4GUI_Caption2FontClr, &::GraphicsResource.TextFont);
```

Comment:
Why did you use the auto keyword here?

Reply:
This way you don't need to repeat the type.

Reply:
Yes, sure, but is that such a problem?

Reply:
Yes. http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-auto